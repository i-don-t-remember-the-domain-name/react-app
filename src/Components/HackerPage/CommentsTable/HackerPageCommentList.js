import React from 'react';
import styled from 'styled-components';
import theme from 'styled-theming';

//Import colors
import { ltLightGray, dtLightBlue } from '../../../colors.js';

//Import components
import CommentRow from './CommentRow';
import CommentTopRow from './CommentTopRow';

//Export default component
export default function HackerPageCommentList(props) {
  return (
    <SDCommentList>
      <CommentTopRow />
      {props.saltiestComments.map(comment => (
        <CommentRow key={comment.comment_id} comment={comment} />
      ))}
    </SDCommentList>
  );
}

//Theming

const backgroundColor = theme('mode', {
  light: ltLightGray,
  dark: dtLightBlue
});

//Styled components
const SDCommentList = styled.div`
  box-sizing: border-box;
  width: 100%;
  min-height: 40vh;
  padding: 2%;
  background-color: ${backgroundColor};
  display: flex;
  flex-direction: column;
  align-items: space-between;
  justify-content: center;
`;
