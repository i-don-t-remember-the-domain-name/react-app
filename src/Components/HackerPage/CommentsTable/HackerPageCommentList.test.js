import React from 'react';

import * as tlr from '@testing-library/react';

import HackerPageCommentList from './HackerPageCommentList';

afterEach(tlr.cleanup);

const saltiestComments = [{ comment_id: 1, comment: 'hello' }, { comment_id: 2, comment: 'hello' }];

describe('Hacker page comment list', () => {
    it('matches the snapshot ', () => {
        const wrap = tlr.render(<HackerPageCommentList saltiestComments={saltiestComments} />);
        expect(wrap).toMatchSnapshot();
    });
});
