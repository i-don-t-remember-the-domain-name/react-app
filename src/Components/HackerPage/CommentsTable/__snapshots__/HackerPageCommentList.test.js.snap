// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Hacker page comment list matches the snapshot  1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="sc-htoDjs gVJxJd"
      >
        <div
          class="sc-ifAKCX bwclxk"
        >
          <div
            class="sc-EHOje fwrCJI"
          >
            Comments
          </div>
          <div
            class="sc-bZQynM lpbUAK"
          >
            Score
          </div>
          <div
            class="sc-gzVnrw imBKBt"
          >
            See on HN
          </div>
        </div>
        <div>
          <div
            class="sc-bdVaJa bjICBm"
          >
            <div
              class="sc-bxivhb iDYKjJ"
            >
              <a
                href="https://news.ycombinator.com/item?id=undefined"
                rel="noopener noreferrer"
                target="_blank"
              >
                <div>
                  <span
                    aria-label="eyes"
                    role="img"
                  >
                    ðŸ‘€
                  </span>
                </div>
              </a>
            </div>
          </div>
        </div>
        <div>
          <div
            class="sc-bdVaJa bjICBm"
          >
            <div
              class="sc-bxivhb iDYKjJ"
            >
              <a
                href="https://news.ycombinator.com/item?id=undefined"
                rel="noopener noreferrer"
                target="_blank"
              >
                <div>
                  <span
                    aria-label="eyes"
                    role="img"
                  >
                    ðŸ‘€
                  </span>
                </div>
              </a>
            </div>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="sc-htoDjs gVJxJd"
    >
      <div
        class="sc-ifAKCX bwclxk"
      >
        <div
          class="sc-EHOje fwrCJI"
        >
          Comments
        </div>
        <div
          class="sc-bZQynM lpbUAK"
        >
          Score
        </div>
        <div
          class="sc-gzVnrw imBKBt"
        >
          See on HN
        </div>
      </div>
      <div>
        <div
          class="sc-bdVaJa bjICBm"
        >
          <div
            class="sc-bxivhb iDYKjJ"
          >
            <a
              href="https://news.ycombinator.com/item?id=undefined"
              rel="noopener noreferrer"
              target="_blank"
            >
              <div>
                <span
                  aria-label="eyes"
                  role="img"
                >
                  ðŸ‘€
                </span>
              </div>
            </a>
          </div>
        </div>
      </div>
      <div>
        <div
          class="sc-bdVaJa bjICBm"
        >
          <div
            class="sc-bxivhb iDYKjJ"
          >
            <a
              href="https://news.ycombinator.com/item?id=undefined"
              rel="noopener noreferrer"
              target="_blank"
            >
              <div>
                <span
                  aria-label="eyes"
                  role="img"
                >
                  ðŸ‘€
                </span>
              </div>
            </a>
          </div>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
